### ISSUE TITLE
"The Dynamic pricing tier is not allowed in resource group" or "LinuxDynamicWorkersNotAllowedInResourceGroup" deploying ARM

### ISSUE THREAD

**Issue by mslot** [Azure/azure-functions-host#6014]
When deploying this ARM from Visual Studio

```arm
{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
  },
  "variables": {
  },
  "resources": [

    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-02-01",
      "name": "test-linux-app-service-plan",
      "location": "[resourceGroup().location]",
      "sku": {
        "name": "S1"
      },
      "kind": "linux",
      "properties": {
        "name": "test-linux-app-service-plan",
        "reserved": true
      }
    },
    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-02-01",
      "name": "test-linux-container-service-plan",
      "location": "[resourceGroup().location]",
      "kind": "linux",
      "properties": {
        "name": "test-linux-container-service-plan",
        "reserved": true
      },
      "sku": {
        "name": "Y1",
        "tier": "Dynamic"
      }

    }
  ],
  "outputs": {
  }
}
```
To a fresh resource group I get one of the following error:

```json
{
    "Code": "BadRequest",
    "Message": "Requested feature is not available in resource group AzureResourceGroup1. Please try using a different resource group or create a new one.",
    "Target": null,
    "Details": [
        {
            "Message": "Requested feature is not available in resource group AzureResourceGroup1. Please try using a different resource group or create a new one."
        },
        {
            "Code": "BadRequest"
        },
        {
            "ErrorEntity": {
                "ExtendedCode": "59324",
                "MessageTemplate": "Requested feature is not available in resource group {0}. Please try using a different resource group or create a new one.",
                "Parameters": [
                    "AzureResourceGroup1"
                ],
                "Code": "BadRequest",
                "Message": "Requested feature is not available in resource group AzureResourceGroup1. Please try using a different resource group or create a new one."
            }
        }
    ],
    "Innererror": null
}
```

I i try to make a dependency, and deploy it

```arm
{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
  },
  "variables": {
  },
  "resources": [

    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-02-01",
      "name": "test-linux-app-service-plan",
      "location": "[resourceGroup().location]",
      "sku": {
        "name": "B1",
        "tier": "Basic"
      },
      "kind": "linux",
      "properties": {
        "name": "test-linux-app-service-plan",
        "reserved": true
      }
    },
    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-11-01",
      "name": "test-linux-container-service-plan",
      "location": "[resourceGroup().location]",
      "kind": "linux",
       "dependsOn": [
        "[resourceId('Microsoft.Web/serverfarms', 'test-linux-app-service-plan')]"
      ],
      "properties": {
        "name": "test-linux-container-service-plan",
        "reserved": true
      },
      "sku": {
        "name": "Y1",
        "tier": "Dynamic"
      }

    }
  ],
  "outputs": {
  }
}
```

I get the above error from before on the first deploy, and when I try to redeploy it, I get this error in my Visual Studio Output pane

> 12:02:20 - Template deployment returned the following errors:
> 12:02:20 - 12:02:19 - Error: Code=InvalidTemplateDeployment; Message=The template deployment 'azuredeploy-0507-1002' is not valid according to the validation procedure. The tracking id is 'c8389566-42ce-43ed-ab54-3c5325d69588'. See inner errors for details.
> 12:02:20 - 12:02:19 - Error: Code=ValidationForResourceFailed; Message=Validation failed for a resource. Check 'Error.Details[0]' for more information.
> 12:02:20 - 12:02:19 - Error: Code=DynamicSkuNotAllowedInResourceGroup; Message=The Dynamic pricing tier is not allowed in resource group AzureResourceGroup1-2. Use this link to learn more http://go.microsoft.com/fwlink/?LinkId=825764.
> 12:02:20 - The deployment validation failed

I have tried to:
*  allign the api version in a fresh resource group, gives the precise same error as above: "requested feature not available" and then error in Output pane in Visual Studio with "DynamicSkuNotAllowed"
* break it up over two deployments, so first we deploy the standard service plan, then the dynamic plan. This still gives same result

If we do this:
1. deploy the dynamic serviceplan to another resource group, and then
2. move it into the resource group already created, with a standard Linux service plan

it works. But this I dont see as a good solution, because it renders the purpose of the ARM template useless. We want to be able to do this automatically.

Is there a workaround for this? Or am I doing something wrong?
Is there a work around on this? We really need to be able to deploy function apps on linux consumption together with.

**EDIT**
If i reverse the dependency order:

```arm
{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
  },
  "variables": {
  },
  "resources": [

    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-02-01",
      "name": "test-linux-app-service-plan",
      "location": "[resourceGroup().location]",
      "sku": {
        "name": "B1",
        "tier": "Basic"
      },
      "kind": "linux",
      "dependsOn": [
        "test-linux-container-service-plan"
      ],
      "properties": {
        "name": "test-linux-app-service-plan",
        "reserved": true
      }
    },
    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-02-01",
      "name": "test-linux-container-service-plan",
      "location": "[resourceGroup().location]",
      "kind": "linux",
      "properties": {
        "name": "test-linux-container-service-plan",
        "reserved": true
      },
      "sku": {
        "name": "Y1",
        "tier": "Dynamic"
      }

    }
  ],
  "outputs": {
  }
}
```
and deploys it to a fresh resource group, it works, but this doesnt quite help us because the whole reason for this is that we want to add a dynamic plan to a resource group that already contains a standard webservice plan. I have tried the following:

**1) first deploy standard service plan to a fresh resource group with this template**
```
{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
  },
  "variables": {
  },
  "resources": [

    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-02-01",
      "name": "test-linux-app-service-plan",
      "location": "[resourceGroup().location]",
      "sku": {
        "name": "B1",
        "tier": "Basic"
      },
      "kind": "linux",
      //"dependsOn": [
      //  "test-linux-container-service-plan"
      //],
      "properties": {
        "name": "test-linux-app-service-plan",
        "reserved": true
      }
    }
    //{
    //  "type": "Microsoft.Web/serverfarms",
    //  "apiVersion": "2018-02-01",
    //  "name": "test-linux-container-service-plan",
    //  "location": "[resourceGroup().location]",
    //  "kind": "linux",
    //  "properties": {
    //    "name": "test-linux-container-service-plan",
    //    "reserved": true
    //  },
    //  "sku": {
    //    "name": "Y1",
    //    "tier": "Dynamic"
    //  }

    //}
  ],
  "outputs": {
  }
}
```
It deploys fine. 

**2) Then I uncomment the dynamic plan, and redeploy this to the same resource group that now has a standard service plan**
```
{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
  },
  "variables": {
  },
  "resources": [

    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-02-01",
      "name": "test-linux-app-service-plan",
      "location": "[resourceGroup().location]",
      "sku": {
        "name": "B1",
        "tier": "Basic"
      },
      "kind": "linux",
      "dependsOn": [
        "test-linux-container-service-plan"
      ],
      "properties": {
        "name": "test-linux-app-service-plan",
        "reserved": true
      }
    },
    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-02-01",
      "name": "test-linux-container-service-plan",
      "location": "[resourceGroup().location]",
      "kind": "linux",
      "properties": {
        "name": "test-linux-container-service-plan",
        "reserved": true
      },
      "sku": {
        "name": "Y1",
        "tier": "Dynamic"
      }

    }
  ],
  "outputs": {
  }
}
```

Then redeploy, and i get this error in output console of Visual Studio

> 12:32:19 - Template deployment returned the following errors:
> 12:32:19 - 12:32:18 - Error: Code=InvalidTemplateDeployment; Message=The template deployment 'azuredeploy-0507-1032' is not valid according to the validation procedure. The tracking id is 'b886c4e9-f5dc-46ee-9e52-cefd970d3623'. See inner errors for details.
> 12:32:19 - 12:32:18 - Error: Code=ValidationForResourceFailed; Message=Validation failed for a resource. Check 'Error.Details[0]' for more information.
> 12:32:19 - 12:32:18 - Error: Code=LinuxDynamicWorkersNotAllowedInResourceGroup; Message=Linux dynamic workers are not available in resource group AzureResourceGroup1-4. Use this link to learn more http://go.microsoft.com/fwlink/?LinkId=825764.
> 12:32:19 - The deployment validation failed

Still not possible :(

Possible workarounds as i see it:
1. first time deploy will fail - then we manually create function app in another resource group, move it and redeploy, and everything will work fine afterwards
2. have a resource group only for linux consumption and a resource group only for linux standard webapps

I prefer number 2). Is there any implications of doing this? As i see it they will still be in same datacenter.

**Comment by balag0** [comment]
To create a function app, I would expect the arm template to include a site resource with a dependency on serverfarm. something like - https://github.com/Azure/azure-quickstart-templates/blob/master/101-app-function/azuredeploy.json

Since you are creating a linux app, setting reserved = true like you have in the template is correct.
Let me know if this doesn't work..

**Comment by mslot** [comment]
> To create a function app, I would expect the arm template to include a site resource with a dependency on serverfarm. something like - https://github.com/Azure/azure-quickstart-templates/blob/master/101-app-function/azuredeploy.json
> 
> Since you are creating a linux app, setting reserved = true like you have in the template is correct.
> Let me know if this doesn't work..

How would that fix the creation of the service plan? As I see it, the service plan need to be in place before a site resource can be spun up, and a site resource can't be spun up because i get the intial error when creating the service plans.

**Comment by balag0** [comment]
Sorry. I misunderstood the question thinking you were looking to create an app + app service plan.

I just tried the template in West US and worked fine for me
![image](https://user-images.githubusercontent.com/28721598/81777948-c2afbb00-94a6-11ea-9ccf-f7ecc30d35f8.png)

I suspect the resource group you are deploying to already has some Microsoft.Web resources which is causing some validation fail. (There are some limitations on what resource types can be in the same resource group)

Do you mind running the template on a empty resource group and see if that works. Let me know if that works.

**Comment by mslot** [comment]
> Sorry. I misunderstood the question thinking you were looking to create an app + app service plan.
> 
> I just tried the template in West US and worked fine for me
> ![image](https://user-images.githubusercontent.com/28721598/81777948-c2afbb00-94a6-11ea-9ccf-f7ecc30d35f8.png)
> 
> I suspect the resource group you are deploying to already has some Microsoft.Web resources which is causing some validation fail. (There are some limitations on what resource types can be in the same resource group)
> 
> Do you mind running the template on a empty resource group and see if that works. Let me know if that works.

I know of the windows/linux limitation. We dont have any windows resources in the ARM nor does it exist in the group before ARM deploy. I have done this on a fresh resource group.

**Comment by balag0** [comment]
Please share the timestamp , region and appservice plan names from when you tried on empty resource group.

**Comment by mslot** [comment]
> Please share the timestamp , region and appservice plan names from when you tried on empty resource group.

```arm
{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
  },
  "variables": {
  },
  "resources": [

    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-02-01",
      "name": "test-linux-app-service-plan",
      "location": "[resourceGroup().location]",
      "sku": {
        "name": "S1"
      },
      "kind": "linux",
      "properties": {
        "name": "test-linux-app-service-plan",
        "reserved": true
      }
    },
    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-02-01",
      "name": "test-linux-container-service-plan",
      "location": "[resourceGroup().location]",
      "kind": "linux",
      "properties": {
        "name": "test-linux-container-service-plan",
        "reserved": true
      },
      "sku": {
        "name": "Y1",
        "tier": "Dynamic"
      }

    }
  ],
  "outputs": {
  }
}
```

Error

> 15:16:44 - Template deployment returned the following errors:
> 15:16:44 - 15:16:43 - Resource Microsoft.Web/serverfarms 'test-linux-container-service-plan' failed with message '{
> 15:16:44 -   "Code": "BadRequest",
> 15:16:44 -   "Message": "Requested feature is not available in resource group rg-test-for-github-issue. Please try using a different resource group or create a new one.",
> 15:16:44 -   "Target": null,
> 15:16:44 -   "Details": [
> 15:16:44 -     {
> 15:16:44 -       "Message": "Requested feature is not available in resource group rg-test-for-github-issue. Please try using a different resource group or create a new one."
> 15:16:44 -     },
> 15:16:44 -     {
> 15:16:44 -       "Code": "BadRequest"
> 15:16:44 -     },
> 15:16:44 -     {
> 15:16:44 -       "ErrorEntity": {
> 15:16:44 -         "ExtendedCode": "59324",
> 15:16:44 -         "MessageTemplate": "Requested feature is not available in resource group {0}. Please try using a different resource group or create a new one.",
> 15:16:44 -         "Parameters": [
> 15:16:44 -           "rg-test-for-github-issue"
> 15:16:44 -         ],
> 15:16:44 -         "Code": "BadRequest",
> 15:16:44 -         "Message": "Requested feature is not available in resource group rg-test-for-github-issue. Please try using a different resource group or create a new one."
> 15:16:44 -       }
> 15:16:44 -     }
> 15:16:44 -   ],
> 15:16:44 -   "Innererror": null
> 15:16:44 - }'

Info form deployment job (with timestamp, corrolation id etc)

> 15:16:43 - DeploymentName          : azuredeploy-0515-1316
> 15:16:43 - CorrelationId           : e9db9b62-d0ae-4a08-86e6-b16502aabc33
> 15:16:43 - ResourceGroupName       : rg-test-for-github-issue
> 15:16:43 - ProvisioningState       : Failed
> 15:16:43 - Timestamp               : 15-05-2020 13:16:40
> 15:16:43 - Mode                    : Incremental
> 15:16:43 - TemplateLink            : 
> 15:16:43 - TemplateLinkString      : 
> 15:16:43 - DeploymentDebugLogLevel : 
> 15:16:43 - Parameters              : {}
> 15:16:43 - ParametersString        : 
> 15:16:43 - Outputs                 : 
> 15:16:43 - OutputsString           : 

Names:
1. S1 plan: test-linux-container-service-plan
2. Dynamic plan: test-linux-app-service-plan

Full id to resource group: /subscriptions/4ec6f523-de5f-4e8a-a6e7-45e4f53170df/resourceGroups/rg-test-for-github-issue

Resource group is created when visual studio is deploying. It is therefore completely new. I have just discovered this: https://github.com/Azure/Azure-Functions/wiki/Creating-Function-Apps-in-an-existing-Resource-Group. Does it have something to do with what I am experiencing? Please note that I get a couple of different errors:

1. Error: Code=DynamicSkuNotAllowedInResourceGroup; Message=The Dynamic pricing tier is not allowed in resource group
2. Error: Code=LinuxDynamicWorkersNotAllowedInResourceGroup; Message=Linux dynamic workers are not available in resource group

**Comment by mslot** [comment]
@balag0 any news on this?

**Comment by brettsam** [comment]
@balag0 -- did you have a chance to look into this?

**Comment by balag0** [comment]
Sorry I must have missed this issue. I tried looking for the request id in the logs but it looks like we are past the log retention period.

Could you please try again and let us know if you are still seeing this issue.  I would expect using an empty resource group to mitigate the issue definitely. So this could be a bug somewhere.

**Comment by mslot** [comment]
I am unable to do it right now, but I will try Wednesday next week, @balag0.

**Comment by iOvergaard** [comment]
@balag0 

I got one here with the same issue.

Tracking ID: `0110b80d-5a41-47e2-946d-f9126f207d52`

```
{
    "Code": "BadRequest",
    "Message": "Linux dynamic workers are not available in resource group . Use this link to learn more http://go.microsoft.com/fwlink/?LinkId=825764.",
    "Target": null,
    "Details": [
        {
            "Message": "Linux dynamic workers are not available in resource group . Use this link to learn more http://go.microsoft.com/fwlink/?LinkId=825764."
        },
        {
            "Code": "BadRequest"
        },
        {
            "ErrorEntity": {
                "ExtendedCode": "59318",
                "MessageTemplate": "Linux dynamic workers are not available in resource group {0}. Use this link to learn more http://go.microsoft.com/fwlink/?LinkId=825764.",
                "Parameters": [
                    ""
                ],
                "Code": "BadRequest",
                "Message": "Linux dynamic workers are not available in resource group . Use this link to learn more http://go.microsoft.com/fwlink/?LinkId=825764."
            }
        }
    ],
    "Innererror": null
}
```

**Comment by balag0** [comment]
@Jacob87 I couldn't find that tracking id in the last 30 days. Would you have the correlation id by any chance?
If not can you share the region and the approximate timestamp to help narrow the search a little.

**Comment by iOvergaard** [comment]
@balag0 Sure thing!

Correlation id: `1d304108-fbe8-4283-a04e-a76206bc599c`
Timestamp: `8/11/2020, 9:20:50 AM`
Region: `West Europe`

I believe it's the same region as @mslot is using.

**Comment by balag0** [comment]
Thanks, Looks like you eventually got it to work with the same region and resource group.

My theory is this was a transient sync error. Each scale unit maintains a list of supported features, skus and this list was not up to date at the time of the error. The list eventually reached a consistent state soon after this failure and future app creations are going through successfully now. This can happen sometimes when a new scale unit is being opened.

Let me know if this is not the case and you are still seeing these errors and I will investigate further.

**Comment by iOvergaard** [comment]
@balag0 Sorry for not explaining further about that resource group. I got it to work by deploying the resources to another resource group and then moving them over to the original resource group manually through the UI, just like @mslot suggested in the OP.

When the resources have been moved manually, any further **incremental** deployments will work.

If I remove the resources and recreate them, I get the same error.

**Comment by mslot** [comment]
@balag0 i will recreate this tomorrow, Friday and send you the I'd. I am not able to do it today. When ever I do a deploy of the ARM template in question in my initial question, I get the error. I will try to do it tomorrow, Friday. I am not able to do it today, sorry.

**Comment by mslot** [comment]
@balag0 I have just tried to deploy this ARM

```json
{
  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
  },
  "variables": {
  },
  "resources": [

    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-02-01",
      "name": "test-linux-app-service-plan",
      "location": "[resourceGroup().location]",
      "sku": {
        "name": "B1",
        "tier": "Basic"
      },
      "kind": "linux",
      "properties": {
        "name": "test-linux-app-service-plan",
        "reserved": true
      }
    },
    {
      "type": "Microsoft.Web/serverfarms",
      "apiVersion": "2018-11-01",
      "name": "test-linux-container-service-plan",
      "location": "[resourceGroup().location]",
      "kind": "linux",
      "dependsOn": [
        "[resourceId('Microsoft.Web/serverfarms', 'test-linux-app-service-plan')]"
      ],
      "properties": {
        "name": "test-linux-container-service-plan",
        "reserved": true
      },
      "sku": {
        "name": "Y1",
        "tier": "Dynamic"
      }

    }
  ],
  "outputs": {
  }
}
```

I got the same error again. Relevant info
1. CorlationId: 025734d6-7be0-42d4-ade8-37fd9c19f2b7
2. Resource group name: rg-linux-dynamic-test
3. Timestamp: 14-08-2020 06:24:08
Location: West Europe

Full error message

> 08:23:54 - Account          : Martin Slot
08:23:54 - SubscriptionName : [removed]
08:23:54 - SubscriptionId   : [removed]
08:23:54 - TenantId         : [removed]
08:23:54 - Environment      : AzureCloud
08:23:54 - 
08:23:57 - VERBOSE: Performing the operation "Creating Deployment" on target "rg-linux-dynamic-test".
08:23:59 - VERBOSE: 08:23:59 - Template is valid.
08:23:59 - VERBOSE: 08:23:59 - Create template deployment 'azuredeploy-0814-0623'
08:23:59 - VERBOSE: 08:23:59 - Checking deployment status in 5 seconds
08:24:05 - VERBOSE: 08:24:05 - Checking deployment status in 5 seconds
08:24:10 - New-AzureRmResourceGroupDeployment : 08:24:10 - Resource Microsoft.Web/serverfarms 'test-linux-container-service-plan' 
08:24:10 - failed with message '{
08:24:10 -   "Code": "BadRequest",
08:24:10 -   "Message": "Requested feature is not available in resource group rg-linux-dynamic-test. Please try using a different 
08:24:10 - resource group or create a new one.",
08:24:10 -   "Target": null,
08:24:10 -   "Details": [
08:24:10 -     {
08:24:10 -       "Message": "Requested feature is not available in resource group rg-linux-dynamic-test. Please try using a differ
08:24:10 - ent resource group or create a new one."
08:24:10 -     },
08:24:10 -     {
08:24:10 -       "Code": "BadRequest"
08:24:10 -     },
08:24:10 -     {
08:24:10 -       "ErrorEntity": {
08:24:10 -         "ExtendedCode": "59324",
08:24:10 -         "MessageTemplate": "Requested feature is not available in resource group {0}. Please try using a different reso
08:24:10 - urce group or create a new one.",
08:24:10 -         "Parameters": [
08:24:10 -           "rg-linux-dynamic-test"
08:24:10 -         ],
08:24:10 -         "Code": "BadRequest",
08:24:10 -         "Message": "Requested feature is not available in resource group rg-linux-dynamic-test. Please try using a diff
08:24:10 - erent resource group or create a new one."
08:24:10 -       }
08:24:10 -     }
08:24:10 -   ],
08:24:10 -   "Innererror": null
08:24:10 - }'
08:24:10 - At C:\Users\mslot\source\repos\rg-linux-dynamic-test\rg-linux-dynamic-test\bin\Debug\staging\rg-linux-dynamic-test\Depl
08:24:10 - oy-AzureResourceGroup.ps1:110 char:5
08:24:10 - +     New-AzureRmResourceGroupDeployment -Name ((Get-ChildItem $Templat ...
08:24:10 - +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
08:24:10 -     + CategoryInfo          : NotSpecified: (:) [New-AzureRmResourceGroupDeployment], Exception
08:24:10 -     + FullyQualifiedErrorId : Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.NewAzureResourceGroupDep 
08:24:10 -    loymentCmdlet
08:24:10 -  
08:24:10 - VERBOSE: 08:24:10 - Resource Microsoft.Web/serverfarms 'test-linux-app-service-plan' provisioning status is succeeded
08:24:10 - New-AzureRmResourceGroupDeployment : 08:24:10 - Template output evaluation skipped: at least one resource deployment op
08:24:10 - eration failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage det
08:24:10 - ails.
08:24:10 - At C:\Users\mslot\source\repos\rg-linux-dynamic-test\rg-linux-dynamic-test\bin\Debug\staging\rg-linux-dynamic-test\Depl
08:24:10 - oy-AzureResourceGroup.ps1:110 char:5
08:24:10 - +     New-AzureRmResourceGroupDeployment -Name ((Get-ChildItem $Templat ...
08:24:10 - +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
08:24:10 -     + CategoryInfo          : NotSpecified: (:) [New-AzureRmResourceGroupDeployment], Exception
08:24:10 -     + FullyQualifiedErrorId : Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.NewAzureResourceGroupDep 
08:24:10 -    loymentCmdlet
08:24:10 -  
08:24:10 - New-AzureRmResourceGroupDeployment : 08:24:10 - Template output evaluation skipped: at least one resource deployment op
08:24:10 - eration failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage det
08:24:10 - ails.
08:24:10 - At C:\Users\mslot\source\repos\rg-linux-dynamic-test\rg-linux-dynamic-test\bin\Debug\staging\rg-linux-dynamic-test\Depl
08:24:10 - oy-AzureResourceGroup.ps1:110 char:5
08:24:10 - +     New-AzureRmResourceGroupDeployment -Name ((Get-ChildItem $Templat ...
08:24:10 - +     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
08:24:10 -     + CategoryInfo          : NotSpecified: (:) [New-AzureRmResourceGroupDeployment], Exception
08:24:10 -     + FullyQualifiedErrorId : Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation.NewAzureResourceGroupDep 
08:24:10 -    loymentCmdlet
08:24:10 -  
08:24:10 - 
08:24:10 - DeploymentName          : azuredeploy-0814-0623
08:24:10 - CorrelationId           : 025734d6-7be0-42d4-ade8-37fd9c19f2b7
08:24:10 - ResourceGroupName       : rg-linux-dynamic-test
08:24:10 - ProvisioningState       : Failed
08:24:10 - Timestamp               : 14-08-2020 06:24:08
08:24:10 - Mode                    : Incremental
08:24:10 - TemplateLink            : 
08:24:10 - TemplateLinkString      : 
08:24:10 - DeploymentDebugLogLevel : 
08:24:10 - Parameters              : {}
08:24:10 - ParametersString        : 
08:24:10 - Outputs                 : 
08:24:10 - OutputsString           : 
08:24:10 - 
08:24:11 - 
08:24:11 - Template deployment returned the following errors:
08:24:11 - 08:24:10 - Resource Microsoft.Web/serverfarms 'test-linux-container-service-plan' failed with message '{
08:24:11 -   "Code": "BadRequest",
08:24:11 -   "Message": "Requested feature is not available in resource group rg-linux-dynamic-test. Please try using a different resource group or create a new one.",
08:24:11 -   "Target": null,
08:24:11 -   "Details": [
08:24:11 -     {
08:24:11 -       "Message": "Requested feature is not available in resource group rg-linux-dynamic-test. Please try using a different resource group or create a new one."
08:24:11 -     },
08:24:11 -     {
08:24:11 -       "Code": "BadRequest"
08:24:11 -     },
08:24:11 -     {
08:24:11 -       "ErrorEntity": {
08:24:11 -         "ExtendedCode": "59324",
08:24:11 -         "MessageTemplate": "Requested feature is not available in resource group {0}. Please try using a different resource group or create a new one.",
08:24:11 -         "Parameters": [
08:24:11 -           "rg-linux-dynamic-test"
08:24:11 -         ],
08:24:11 -         "Code": "BadRequest",
08:24:11 -         "Message": "Requested feature is not available in resource group rg-linux-dynamic-test. Please try using a different resource group or create a new one."
08:24:11 -       }
08:24:11 -     }
08:24:11 -   ],
08:24:11 -   "Innererror": null
08:24:11 - }'
08:24:11 - 08:24:10 - Template output evaluation skipped: at least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.
08:24:11 - 08:24:10 - Template output evaluation skipped: at least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.
08:24:11 - 
08:24:11 - 
08:24:11 - Deploying template using PowerShell script failed.
08:24:11 - Tell us about your experience at https://go.microsoft.com/fwlink/?LinkId=691202

**Comment by mslot** [comment]
@balag0 do you have a status on this?

**Comment by balag0** [comment]
@mslot sorry missed the update on this issue.

I looked at this failure and here is what I found
The resource group you are using "rg-linux-dynamic-test" in West Europe doesn't have any preexisting function app in it, but it does have an app service plan already named - "test-linux-app-service-plan"

Internally function apps are provisioned on different scale units across different regions. When a new function app / web app / server farm is created in a completely empty resource group (from the point of view of Microsoft.Web RP i.e no web apps, server farms, functions apps etc, and storage accounts, event grids etc don't count) a random scale unit is selected in the region that supports all the required features. The resource group is now tied to that scale unit. (meaning any other function app / web app you create in that region will always land on that scale unit)

In your case when you created 'test-linux-app-service-plan ' server farm the required features were Linux and App service plan function app. The scale unit selection was based on these requirements. Note that this scale unit doesn't have to support dynamic sku since the server farm didn't request that feature.

When the above arm template you shared tried to create a dynamic function app in the same resource group + same region, an attempt was made to create the app in the scale unit selected previously. Since that scale unit doesn't support dynamic sku, you get a conflict. 

Workarounds - use a different resource group (either empty or containing linux dynamic function apps only)
Use a different region. Since the scale unit selection also considers the region, and the preexisting resource group is in a different region, the new app will not be placed in the same scale unit and scale unit selection will take place again.

We understand this is confusing and we are working on allowing apps from different scale units in same region within a single resource group. Once this is possible we can make the scale unit selection for creation of each resource instead of the current implementation where the selection takes place only for the 1st resource in RG.
